### PR TITLE
Get client properties from ServiceClient instead of OperationArguments

### PR DESCRIPTION
**Pull request by daschult**
Resolves https://github.com/Azure/autorest.typescript/issues/189.

### CODE REVIEW

File: lib/serviceClient.ts [hunk 1]
```diff
@@ -384,25 +384,23 @@ export function getPropertyParent(parent: PropertyParent, propertyPath: string[]
...
   if (operationArguments.arguments) {
     if (Array.isArray(parameterPath)) {
       if (parameterPath.length > 0) {
-        value = operationArguments.arguments;
-        for (const parameterPathPart of parameterPath) {
-          value = value[parameterPathPart];
-          if (value == undefined) {
-            break;
-          }
+        let propertySearchResult: PropertySearchResult = getPropertyFromParameterPath(operationArguments.arguments, parameterPath);
```
Comment (RikkiGibson) [inline lib/serviceClient.ts]: This feels like a nice place for `const { propertyFound, propertyValue } = ...`

File: lib/serviceClient.ts [hunk 2]
```diff
@@ -427,3 +425,27 @@ function getOperationArgumentValueFromParameterPath(operationArguments: Operatio
...
+  propertyValue?: any;
+  propertyFound: boolean;
+}
+
+function getPropertyFromParameterPath(parent: { [parameterName: string]: any }, parameterPath: string[]): PropertySearchResult {
+  const result: PropertySearchResult = { propertyFound: false };
+  let i = 0;
+  for (; i < parameterPath.length; ++i) {
+    const parameterPathPart: string = parameterPath[i];
+    // Make sure to check inherited properties too, so don't use hasOwnProperty().
```
Comment (RikkiGibson) [inline lib/serviceClient.ts]: `in parent` means that the parent contains a property with specifically that name?

Comment (daschult) [reply lib/serviceClient.ts]: Yup. That's what I want.

File: lib/serviceClient.ts (hunk 1 above)
Comment (daschult) [reply lib/serviceClient.ts]: Can't do that here, because I may reassign propertySeachResult in two lines.

**Pull request merged**